runtime/caml/misc.h

```diff
@@ -113,6 +113,16 @@ CAMLdeprecated_typedef(addr, char *);
 #error "How do I align values on this platform?"
 #endif
 
+/* Prefetching */
+
+#ifdef CAML_INTERNALS
+#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#define caml_prefetch(p) __builtin_prefetch((p), 1, 3)
```

Comment:
I think - but I have not experimented - that the MSVC equivalent is `#include <winnt.h>` and `PreFetchCacheLine((p), PF_NON_TEMPORAL_LEVEL_ALL)` (I'm not sure about the constant)

Reply:
Just curious: why only on x86?  `__builtin_prefetch` exists on all GCC-supported platforms, even though it can be a no-op.  And I'm sure ARM and others would benefit too.

Reply:
Also, I would document a bit:
```
#define caml_prefetch(p) __builtin_prefetch((p), 1, 3)
/* 1 = intent to write; 3 = all cache levels */
```

Reply:
> Just curious: why only on x86? __builtin_prefetch exists on all GCC-supported platforms, even though it can be a no-op. And I'm sure ARM and others would benefit too.

I left others out because I don't really know anything about non-x86 memory hierarchies. We can turn it on for ARM if you like, but I can't judge how much / whether it'll help, and I don't have the expertise / time to do any serious benchmarking.

(I'll add the comments)